include/fiction/algorithms/physical_design/hexagonalization.hpp

```diff
@@ -45,9 +47,7 @@ class hexagonalization_io_pin_routing_error : public std::runtime_error
      *
      * @param msg The error message describing the error.
      */
-    explicit hexagonalization_io_pin_routing_error(const std::string_view& msg) noexcept :
-            std::runtime_error(msg.data())
-    {}
+    explicit hexagonalization_io_pin_routing_error(std::string_view msg) : std::runtime_error(std::string{msg}) {}
```

Comment:
Even if you don't want to pass the `std::string_view` by reference, you can still make it `const`.